src/shapes/audio_token.class.js

```diff
@@ -0,0 +1,202 @@
+     * @type number
+     * @default
+     */
+    playbackMS: 0,
+
+    /**
+     * Audio URL? WIP - not sure how this will be used
+     * To be determined by: NEWXP-1949, likely passed in as a parameter on creation
+     * @type string
+     * @default
+     */
+    audioUrl: '',
+
+    hasBorders: false,
+
+    /**
+     * extra space taken up on the left by delete controls (used for collision detection)
+     * @type Number
+     * @default
+     */
+    leftMargin: 44,
+
+    /**
+     * Definable images for idle state, selected state, and play/pause controls
+     */
+    idleImage: null,
+    selectedImage: null,
+    playControlImage: null,
+    pauseControlImage: null,
+
+    idleImageSrc: undefined,
+    selectedImageSrc: undefined,
+    playControlImageSrc: undefined,
+    pauseControlImageSrc: undefined,
+
+    /**
+     * List of properties to consider when checking if
+     * state of an object is changed ({@link fabric.Object#hasStateChanged})
+     * @type Array
+     */
+    stateProperties: fabric.Object.prototype.stateProperties.concat('isPlaying', 'isPaused'),
+
+    /**
+     * Constructor
+     * @param {Object} [options] Options object
+     * @return {fabric.Audio_token} thisArg
+     */
+    initialize: function(audioURL, options) {
+      options || (options = { });
+      this.filters = [];
+      this.audioURL = audioURL;
+      this.cacheKey = 'audio_token' + fabric.Object.__uid++;
+      this.callSuper('initialize', options);
+      this.initBehavior();
+      this.initImages();
+    },
+
+    /**
+     * Draws a background for the object big as its untransformed dimensions
+     * @private
+     * @param {CanvasRenderingContext2D} ctx Context to render on
+     */
+    _renderBackground: function() {
+      return;
+    },
+
+    _isSelected: function() {
+      return this.canvas && this.canvas._activeObject === this;
+    },
+
+    initImages: function() {
+      if (this.idleImageSrc) {
+        //TODO: will this create multiple images if we have multiple instances? Or is this a property of the prototype?
+        this.idleImage = document.createElement('img');
+        this.idleImage.src = this.idleImageSrc;
+      }
+      if (this.selectedImageSrc) {
+        this.selectedImage = document.createElement('img');
+        this.selectedImage.src = this.selectedImageSrc;
+      }
+      if (this.pauseControlImageSrc) {
+        this.pauseControlImage = document.createElement('img');
+        this.pauseControlImage.src = this.pauseControlImageSrc;
+      }
+      if (this.playControlImageSrc) {
+        this.playControlImage = document.createElement('img');
+        this.playControlImage.src = this.playControlImageSrc;
+      }
+    },
+
+    /**
+     * Returns object representation of an instance
+     * @method toObject
+     * @param {Array} [propertiesToInclude] Any properties that you might want to additionally include in the output
+     * @return {Object} object representation of an instance
+     */
```

Comment:
We probably don't need the playstate here? (these additional properties get serialized into the save file)